ospfd/ospf_vty.c

```diff
@@ -5138,19 +5138,36 @@ static void show_ip_ospf_neighbor_detail_sub(struct vty *vty,
 		json_object_string_add(json_neigh, "areaId",
 				       ospf_area_desc_string(oi->area));
 		json_object_string_add(json_neigh, "ifaceName", oi->ifp->name);
-	} else
-		vty_out(vty, "    In the area %s via interface %s\n",
+		if (oi->address)
+			json_object_string_addf(json_neigh, "localIfaceAddress",
+						"%pI4",
+						&oi->address->u.prefix4);
+	} else {
+		vty_out(vty, "    In the area %s via interface %s",
 			ospf_area_desc_string(oi->area), oi->ifp->name);
+		if (oi->address)
+			vty_out(vty, " local interface IP %pI4\n",
+				&oi->address->u.prefix4);
+		else
+			vty_out(vty, "\n");
```

Comment:
Could you show how the output looks with and without local interface IP (with this change)?

Reply:
Before code change:
```
r1# show ip ospf neighbor 10.0.255.2 detail 
 Neighbor 10.0.255.2, interface address 10.0.3.2
    In the area 0.0.0.0 via interface r1-eth1
    Neighbor priority is 1, State is Full/-, 5 state changes
    Most recent state change statistics:
      Progressive change 48.112s ago
    DR is 0.0.0.0, BDR is 0.0.0.0
    Options 2 *|-|-|-|-|-|E|-
    Dead timer due in 8.802s
    Database Summary List 0
    Link State Request List 0
    Link State Retransmission List 0
    Thread Inactivity Timer on
    Thread Database Description Retransmision off
    Thread Link State Request Retransmission on
    Thread Link State Update Retransmission on

    Graceful restart Helper info:
      Graceful Restart HELPER Status : None


r1# 
```

After code change:
```
r1# show ip ospf neighbor 10.0.255.2 detail
 Neighbor 10.0.255.2, interface address 10.0.3.2
    In the area 0.0.0.0 via interface r1-eth1 local interface IP 10.0.3.4
    Neighbor priority is 1, State is Full/-, Role is DROther, 5 state changes
    Most recent state change statistics:
      Progressive change 15.555s ago
    DR is 0.0.0.0, BDR is 0.0.0.0
    Options 2 *|-|-|-|-|-|E|-
    Dead timer due in 8.893s
    Database Summary List 0
    Link State Request List 0
    Link State Retransmission List 0
    Thread Inactivity Timer on
    Thread Database Description Retransmision off
    Thread Link State Request Retransmission on
    Thread Link State Update Retransmission on

    Graceful restart Helper info:
      Graceful Restart HELPER Status : None
```

r1-eth1's addr is 10.0.3.4/32 as shown below
```
r1# show interface brief 
Interface       Status  VRF             Addresses
---------       ------  ---             ---------
lo              up      default         
r1-eth0         up      default         10.0.1.1/32
r1-eth1         up      default         10.0.3.4/32

```

Reply:
Is that double new empty lines in the output?

Reply:
No, those double empty line is existing code. I have not modified it.